debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSchema.java

```diff
@@ -56,7 +57,7 @@
     private final Schema sourceSchema;
     private final SchemaNameAdjuster adjuster = SchemaNameAdjuster.create(LOGGER);
     private final Function<Document, String> valueTransformer;
-    private final Map<CollectionId, MongoDbCollectionSchema> collections = new HashMap<>();
+    private final Map<CollectionId, MongoDbCollectionSchema> collections = new ConcurrentHashMap<>();
```

Comment:
Can you change the variable type to `ConcurrentInterface` to and also see whether some of our marker annotations like `@ThreadSafe`, `Immutable` etc. can be added? Thx!